controllers/users/populateUser.js

```diff
@@ -40,7 +50,12 @@ module.exports = async (req, res) => {
         is_banned,
         users.is_anonymous,
         users.allow_anon_dm
-        from users INNER JOIN user_follow_user ON users.user_id = user_follow_user.user_id_follower WHERE users.user_id != :userId AND user_follow_user.user_id_followed = :userId AND users.is_anonymous = false
+        from users INNER JOIN user_follow_user ON users.user_id = user_follow_user.user_id_follower 
+        WHERE 
+          users.user_id != :userId 
+          AND user_follow_user.user_id_followed = :userId 
+          AND users.is_anonymous = false
+          ${filterAllowDm}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Again, avoid direct string interpolation of user-provided values into SQL queries to prevent SQL injection attacks.

```diff
-    AND users.is_anonymous = false
-    ${filterAllowDm}
+    AND users.is_anonymous = false`,
+    { replacements: { userId: userId, ...filterAllowDm }, type: sequelize.QueryTypes.SELECT }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->